.gitignore

```diff
@@ -36,3 +36,8 @@ node_modules
 
 # Optional REPL history
 .node_repl_history
+
+# Only apps should have lockfiles
+npm-shrinkwrap.json
+package-lock.json
+yarn.lock
```

Comment:
It can be useful for a library to have a lockfile checked in (and not published) so that build and test dependencies are locked to known-good versions.

Reply:
I don’t agree; that allows the package to pass tests while consumers fail them. Dev-only lockfiles are only appropriate for apps, not for packages.

Reply:
I mainly mention because I feel like I've had build and test specific dev-dependencies (such as babel-plugin-add-module-exports today :P ) break much more often than regular dependencies. It's a very painful experience to clone an old repository to make a quick tweak and find it can't even build. I find it important to guarantee that a project is in a buildable (and hopefully even fully-reproducible) state as it was in the last commit. Finding out about breakages in non-dev-dependencies is important, but omitting a lockfile so the next repo-cloner runs into it isn't the specific way I like to go about it for my own projects. (Personally, I like using greenkeeper for that sort of thing.)

I'm happy with whichever way this project goes, just wanted to dump some of my experiences and reasoning here! The main meat of this PR seems really needed and I definitely wouldn't want to delay that.

Reply:
Indeed you should use something like greenkeeper to keep up with that, rather than letting a project bitrot until you try to maintain it later.